lib/adapters/binance.py

```diff
@@ -0,0 +1,166 @@
+from typing import Tuple
+from lib.adapters.base import Adapter, Credentials, InvalidPayloadRecieved
+from lib.account import Account
+from logging import getLogger
+from lib.strategy import Strategy
+from binance.client import Client
+from binance.websockets import BinanceSocketManager
+from lib.account import (
+    AccountStatus,
+    BalanceUpdate,
+    Position, 
+    PositionStatus,
+    UpdateType
+)
+
+
+logger = getLogger()
+
+
+class BinanceAdapter(Adapter):
+    """ """
+
+    class BinanceMarketStreamTransformer:
+        def __call__(self, payload) -> pd.Dataframe:
```

Comment:
What if I don't wish to reduce computation, and don't wish to use a `pandas.DataFrame` as the output?

1. Provide a `use_pandas=bool` flag during instantiation;
2. Define a `TypedDict` wrapper for raw data - i.e. `RawTick`;
3. Return a `Union` type - i.e. `Union[pd.DataFrame, RawTick]`